packages/manager/apps/web/client/app/hosting/hosting.controller.js

```diff
@@ -854,50 +854,25 @@ export default class {
    * The isUpgradable variable can also removed
    * @returns {Promise<{}>}
    */
-  async simulateUpgradeAvailability() {
-    try {
-      const cdnVersion = get(this.$scope.cdnProperties, 'version', '');
+  simulateUpgradeAvailability() {
+    const cdnVersion = get(this.$scope.cdnProperties, 'version', '');
+    if (cdnVersion === HOSTING_CDN_ORDER_CDN_VERSION_V1) {
       const { serviceName } = this.$scope.hosting;
-      if (cdnVersion === HOSTING_CDN_ORDER_CDN_VERSION_V1) {
-        const {
-          data: servInfo,
-        } = await this.HostingCdnSharedService.getServiceInfo(serviceName);
-
-        const {
-          data: servOpts,
-        } = await this.HostingCdnSharedService.getServiceOptions(
-          servInfo.serviceId,
-        );
-
-        const { serviceId } = find(
-          servOpts,
-          ({ billing }) =>
-            billing.plan.code.match('^cdn') &&
-            billing.plan.code.match('_business$'),
-        );
-
-        const {
-          data: addonPlans,
-        } = await this.HostingCdnSharedService.getCatalogAddonsPlan(serviceId);
-
-        const addonPlan = find(addonPlans, ({ planCode }) =>
-          includes(['cdn-basic', 'cdn-basic-free'], planCode),
-        );
-
-        await this.HostingCdnOrderService.simulateCartForUpgrade(
-          serviceName,
-          addonPlan,
-          serviceId,
-        );
-        this.$scope.isUpgradable = true;
-        return {};
-      }
-      this.$scope.isUpgradable = true;
-      return {};
-    } catch (e) {
-      this.$scope.isUpgradable = false;
-      return {};
+      return this.HostingCdnSharedService.simulateUpgrade(
+        serviceName,
+        this.HostingCdnOrderService,
+      )
+        .then(() => {
+          this.$scope.isUpgradable = true;
+          return {};
```

Comment:
you are returning an empty object in each case
i don't think you need to return anything right ?

Reply:
i do that to allow the chaining promise after my call.
if someone add a new call after mine it will works see [this line number](https://github.com/ovh/manager/blob/d7111bc07e7dfcdbcb3de3b08ded21f27ae02fb9/packages/manager/apps/web/client/app/hosting/hosting.controller.js#L827).

Reply:
you can still chain promise if nothing is returned

Reply:
Right i just tested.
May be i confused with something i read.

Reply:
i removed the empty object.